python/test/gluon/test_frontend.py

```diff
@@ -1214,6 +1221,31 @@ def test_async_copy(fresh_knobs):
 """)
 
 
+def from_ast_to_module(fn, device, num_warps, num_ctas):
```

Comment:
Instead of passing in the `device` and hardcoding just two targets here, we should directly pass in the `GPUTarget` as a parameter to this function--if we think about potential future use cases and we will have the needs to test various architectures for the same vendor.